src/lib/crypto/ecdh.c

```diff
@@ -215,7 +215,7 @@ pgp_ecdh_encrypt_pkcs5(const uint8_t *const     session_key,
     uint8_t *       tmp_buf = NULL;
     uint8_t         kek[32] = {0}; // Size of SHA-256 or smaller
 
-    if ((session_key_len > OBFUSCATED_KEY_SIZE) || !ephemeral_key || !pubkey || !wrapped_key ||
+    if ((session_key_len > OBFUSCATED_KEY_SIZE) || !ephemeral_key->mp || !pubkey || !wrapped_key ||
```

Comment:
I think we should be checking both here - "!ephemeral_key || !ephemeral_key->mp"